movies-hub/src/pages/Login/LoginPage.jsx

```diff
@@ -1,117 +1,74 @@
-    if (!password) {
-      setErrors((prev) => ({
-        ...prev,
-        password: "Please enter your password.",
-      }));
-      hasError = true;
-    }
+  const setToken = useAuthStore((state) => state.setToken);
 
-    if (hasError) return;
-
-    try {
-      setLoading(true);
-      const payload = { email, password };
-      const response = await LOGIN_USER_API.login(payload);
-      const data = response.data;
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm();
 
+  const loginMutation = useMutation({
+    mutationFn: (data) => AUTH_API.login(data),
+    onSuccess: () => {
       successToast("Login successful!");
-      setEmail("");
-      setPassword("");
-      localStorage.setItem("token", API_TOKEN);
+      reset();
+      setToken(API_TOKEN);
       navigate(ROUTES.MOVIES);
-    } catch (error) {
+    },
+    onError: (error) => {
       console.error("Login failed:", error);
       errorToast("Invalid credentials. Please try again.");
-    } finally {
-      setLoading(false);
-    }
-  };
+    },
+  });
 
-  const handleForgetPassword = () => {
-    alert("Forgot password functionality not implemented yet.");
+  const onSubmit = (data) => {
+    loginMutation.mutate(data);
   };
 
   return (
     <div className="min-h-screen flex items-center justify-center bg-neutral-700 px-4">
-      <form
-        onSubmit={handleSubmit}
-        className="bg-neutral-800 p-8 rounded-lg shadow-lg flex flex-col w-full max-w-sm"
-      >
-        <h2 className="text-2xl font-bold mb-4 text-white">Login</h2>
-
-        <div className="mb-3">
-          <input
-            id="email"
-            type="email"
-            name="email"
-            placeholder="Email"
-            value={email}
-            onChange={(e) => setEmail(e.target.value)}
-            className="px-3 py-2 rounded bg-neutral-700 border border-neutral-600 text-white focus:outline-none focus:ring-2 focus:ring-red-600 w-full"
-          />
-          {errors.email && (
-            <p className="text-red-500 text-sm mt-1">{errors.email}</p>
-          )}
-        </div>
-
-        <div className="relative mb-3">
-          <input
-            id="password"
-            type={showPassword ? "text" : "password"}
-            name="password"
-            placeholder="Password"
-            value={password}
-            onChange={(e) => setPassword(e.target.value)}
-            className="px-3 py-2 rounded bg-neutral-700 border border-neutral-600 text-white focus:outline-none focus:ring-2 focus:ring-red-600 w-full pr-10"
+      <FormWrapper title="Login" onSubmit={handleSubmit(onSubmit)}>
+        {LOGIN_INPUT_FIELDS.map(({ name, type, placeholder, validation }) => (
+          <InputField
+            key={name}
+            name={name}
+            type={type}
+            placeholder={placeholder}
+            register={register}
+            validation={validation}
+            error={errors[name]}
           />
```

Comment:
update structure for any input field type as I told
login --> inputFields --> [spcific field]